lib/std/Thread.zig

```diff
@@ -70,6 +70,181 @@ else switch (std.Target.current.os.tag) {
 
 pub const spinLoopHint = @compileError("deprecated: use std.atomic.spinLoopHint");
 
+pub const maxNameLen = switch (std.Target.current.os.tag) {
+    .linux => 15,
+    .windows => 31,
+    .macos, .ios, .watchos, .tvos => 63,
+    .netbsd => 31,
+    .freebsd => 15,
+    .openbsd => 31,
+    else => 0,
+};
+
+pub fn setName(self: Thread, name: []const u8) !void {
```

Comment:
This has different errors across different targets and so it must have an explicit error set (e.g. so that "Unsupported" shows up even on targets where it is always supported).

Reply:
Addressed in the latest commit